File: Algorithm-CPP/49_블록의최댓값.cpp
#include <iostream>
using namespace std;

int main() {
    int n;
    cin >> n;

    int front[n][n], side[n][n], res = 0;
    for(int i = 0; i < n; i++) {
        int input;
        cin >> input;
        for(int j = 0; j < n; j++) {
            front[j][i] = input;
        }
    }

    for(int i = 0; i < n; i++) {
        int input;
        cin >> input;
        for(int j = 0; j < n; j++) {
            side[i][j] = input;
        }
    }    

    for(int i = 0; i < n; i++) {
        for(int j = 0; j < n; j++) {
            res += min(front[i][j], side[i][j]);
        }
    }

    cout << res << endl;

    return 0;
}

File: Algorithm-Swift/Algorithm-Swift/Inflearn/49_블록의최댓값.swift
//
//  49_블록의최댓값.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/08/24.
//

import Foundation

func solution49() {
    let n = Int(readLine()!)!
    let front = readLine()!.split(separator: " ").map { Int($0)! }
    let side = readLine()!.split(separator: " ").map { Int($0)! }
    var res = 0

    for i in 0..<n {
        for j in 0..<n {
            res += min(front[j], side[i])
        }
    }

    print(res)
}
